File: 27-Interpreter/src/go/context.go
package main

// Context holds the boolean value bound to each variable name.
type Context struct {
	values map[string]bool
}

func NewContext() *Context {
	return &Context{values: make(map[string]bool)}
}

func (c *Context) Set(name string, value bool) {
	c.values[name] = value
}

func (c *Context) Get(name string) bool {
	return c.values[name]
}

File: 27-Interpreter/src/rust/context.rs
use std::collections::HashMap;

/// Context — holds the boolean value bound to each variable name.
pub struct Context {
    values: HashMap<String, bool>,
}

impl Context {
    pub fn new() -> Self {
        Context { values: HashMap::new() }
    }

    pub fn set(&mut self, name: &str, value: bool) {
        self.values.insert(name.to_string(), value);
    }

    pub fn get(&self, name: &str) -> bool {
        *self.values.get(name).unwrap_or(&false)
    }
}
